typescript/hx-adl/src/gen-sqlschema.ts

```diff
@@ -72,128 +86,185 @@ export async function generateSql(params: Params): Promise<void> {
-      for(const f of t.struct.value.fields) {
+    function findColName(s: string): string {
+      for (const f of t.struct.value.fields) {
         if (f.name == s) {
           return getColumnName(f);
         }
       }
       return s;
     }
 
-    for(let i = 0; i < indexes.length; i++) {
+    for (let i = 0; i < indexes.length; i++) {
       const cols = indexes[i].map(findColName);
-      constraints.push(`create index ${t.name}_${i+1}_idx on ${quoteReservedName(t.name)}(${cols.join(', ')});`);
+      constraints.push(`create index if not exists ${t.name}_${i + 1}_idx on ${quoteReservedName(t.name)}(${cols.join(', ')});`);
     }
-    for(let i = 0; i < uniquenessConstraints.length; i++) {
+    for (let i = 0; i < uniquenessConstraints.length; i++) {
       const cols = uniquenessConstraints[i].map(findColName);
-      constraints.push(`alter table ${quoteReservedName(t.name)} add constraint ${t.name}_${i+1}_con unique (${cols.join(', ')});`);
+      constraints.push(`alter table ${quoteReservedName(t.name)} add constraint ${t.name}_${i + 1}_con unique (${cols.join(', ')});`);
     }
     if (withIdPrimaryKey) {
-      lines.push({code:'primary key(id)'});
+      lines.push({ code: 'primary key(id)' });
+      lines.push({ code: `alter table ${quoteReservedName(t.name)} add primary key(id);` });
     } else if (withPrimaryKey.length > 0) {
       const cols = withPrimaryKey.map(findColName);
-      lines.push({code:`primary key(${cols.join(',')})`});
+      lines.push({ code: `primary key(${cols.join(',')})` });
+      lines.push({ code: `alter table ${quoteReservedName(t.name)} add primary key(${cols.join(',')});` });
     }
 
-
     writer.write('\n');
-    writer.write( `create table ${quoteReservedName(t.name)}(\n` );
-    for(let i = 0; i < lines.length; i++) {
+    writer.write(`create table if not exists ${quoteReservedName(t.name)} ();` + '\n');
+    for (let i = 0; i < lines.length; i++) {
       let line = lines[i].code;
-      if (i < lines.length-1) {
+      if (i < lines.length - 1) {
         line += ',';
+        line = `alter table ${quoteReservedName(t.name)} add column if not exists ${line};`;
+        writer.write(line + '\n');
       }
       if (lines[i].comment) {
         line = line.padEnd(36, ' ');
         line = line + " -- " + lines[i].comment;
       }
       writer.write('  ' + line + '\n');
     }
-    writer.write( `);\n` );
+    writer.write(`);\n`);
+
+    for (let i = 0; i < pkLines.length; i++) {
+      let line = pkLines[i];
+      line = `${line};`;
+      writer.write(line + '\n');
+    }
+
     allExtraSql = allExtraSql.concat(extraSql);
   }
+  writer.write("\ncommit;\n\n");
+  await writer.close();
 
-  if(constraints.length > 0) {
+  if (constraints.length > 0) {
     writer.write('\n');
   }
 
-  for(const constraint of constraints) {
+  writer = fs.createWriteStream(params.outfiles[1]);
+  writer.write("begin;\n\n");
+  writer.write('-- vv START MANUAL CODE FOR MIGRATION' + '\n\n');
```

Comment:
What's the point of this?

(A generated file that says START MANUAL CODE FOR MIGRATION here?)